org.kryogenix.utm_no.yml

```diff
@@ -0,0 +1,53 @@
+app-id: org.kryogenix.utm_no
+runtime: org.gnome.Platform
+runtime-version: '42'
+sdk: org.gnome.Sdk
+command: utm_no
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --talk-name=org.kde.StatusNotifierWatcher
+
+rename-desktop-file: utm_no.desktop
+rename-appdata-file: utm_no.metainfo.xml
+rename-icon: utm_no
+
+modules:
+  - name: utm_no
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 flatpak/launcher.sh /app/bin/utm-no
+      - install -Dm755 flatpak/launcher.sh /app/bin/utm_no
+      - cp -r utm_no /app/utm_no
+      - rm -rf /app/utm_no/__pycache__
+      - install -D utm_no/icons/utm_no.svg /app/share/icons/hicolor/scalable/apps/utm_no.svg
+      - install -D snap/gui/utm_no.desktop /app/share/applications/utm_no.desktop
+      - install -D flatpak/utm_no.metainfo.xml /app/share/metainfo/utm_no.metainfo.xml
+    sources:
+      - type: git
+        url: https://github.com/stuartlangridge/utm_no.git
+        commit: da2fdd785789dd509993c7925209cb57ef591a34
+
+  - name: python3-requests
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "requests" --no-build-isolation
+    sources:
+    - type: file
+      url: https://files.pythonhosted.org/packages/04/a2/d918dcd22354d8958fe113e1a3630137e0fc8b44859ade3063982eacd2a4/idna-3.3-py3-none-any.whl
+      sha256: 84d9dd047ffa80596e0f246e2eab0b391788b0503584e8945f2368256d2735ff
+    - type: file
+      url: https://files.pythonhosted.org/packages/06/b3/24afc8868eba069a7f03650ac750a778862dc34941a4bebeb58706715726/charset_normalizer-2.0.12-py3-none-any.whl
+      sha256: 6881edbebdb17b39b4eaaa821b438bf6eddffb4468cf344f09f89def34a8b1df
+    - type: file
+      url: https://files.pythonhosted.org/packages/2d/61/08076519c80041bc0ffa1a8af0cbd3bf3e2b62af10435d269a9d0f40564d/requests-2.27.1-py2.py3-none-any.whl
+      sha256: f22fa1e554c9ddfd16e6e41ac79759e17be9e492b3587efa038054674760e72d
+    - type: file
+      url: https://files.pythonhosted.org/packages/ec/03/062e6444ce4baf1eac17a6a0ebfe36bb1ad05e1df0e20b110de59c278498/urllib3-1.26.9-py2.py3-none-any.whl
+      sha256: 44ece4d53fb1706f667c9bd1c648f5469a2ec925fcf3a776667042d645472c14
+    - type: file
+      url: https://files.pythonhosted.org/packages/37/45/946c02767aabb873146011e665728b680884cd8fe70dde973c640e45b775/certifi-2021.10.8-py2.py3-none-any.whl
+      sha256: d62a0163eb4c2344ac042ab2bdf75399a71a2d8c7d47eac2e2ee91b9d6339569
+
+  - shared-modules/libappindicator/libappindicator-gtk3-introspection-12.10.json
```

Comment:
you need to add shared-modules as a git submodule. https://github.com/flathub/shared-modules/

Reply:
it's in THIS repository you want the submodule.

Reply:
I feel very stupid, here, because I don't understand what I'm doing wrong. I can check out my repository completely from fresh and then build it and it builds fine, but clearly the buildbot is doing something different, and I don't know what:

```bash
$ git clone --recurse-submodules https://github.com/stuartlangridge/utm_no.git
(it clones ok)
$ cd utm_no/
$ flatpak run org.flatpak.Builder flatpak-build --user --install --force-clean flatpak/org.kryogenix.utm_no.yml
(it builds and installs ok)
```

so I must be missing something very obvious, and apologies for having to have it explained, but I don't get what I'm not doing...

Reply:
as I said: THIS repository, the one with the manifest (it's https://github.com/stuartlangridge/flathub/ ). Not the upstream code.

See the README at https://github.com/flathub/shared-modules/